core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/commandqueue/handlers/TransferProcessCommandHandler.java

```diff
@@ -0,0 +1,31 @@
+package org.eclipse.dataspaceconnector.transfer.core.commandqueue.handlers;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.transfer.command.CommandHandler;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.commands.TransferProcessCommand;
+
+import static java.lang.String.format;
+
+public abstract class TransferProcessCommandHandler<T extends TransferProcessCommand> implements CommandHandler<T> {
+    protected final TransferProcessStore store;
+
+    public TransferProcessCommandHandler(TransferProcessStore store) {
+        this.store = store;
+    }
+
+    @Override
+    public void handle(TransferProcessCommand command) {
+        var transferProcessId = command.getTransferProcessId();
+        var transferProcess = store.find(transferProcessId);
+        if (transferProcess == null) {
+            throw new EdcException(format("Could not find TransferProcess with ID [%s]", transferProcessId));
+        } else {
+            modify(transferProcess);
+            store.update(transferProcess);
```

Comment:
What happens in the case that the implementation of modify(..) does not change the TP? Should it still be saved regardless or can modify return a boolean indicating that the TP was updated?

Reply:
Sounds a bit strange to have a `TransferProcessCommand` that does not actually modify anything, but I added a boolean return value to indicate the need for an update.